### PR TITLE
Custom jquery

### PR DESCRIPTION
**Pull request by Lapochka15**

### CODE REVIEW

File: Task7/src/index.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        return this;
+    }
+
+    static click(node, callback){
+        let element = document.querySelector(node);
+        element.addEventListener('click', callback);
+    };
+
+    foreach(callback){
+        this.foreach(this._string, callback);
```
Comment (alexeypt) [inline Task7/src/index.js]: It must be `jquery.foreach()`

File: Task7/src/index.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            child.parentNode.removeChild( child);   
+        }        
+    };
+
+    css(property){
+        jQuery.css(this._string, property);
+        return this;
+    }
+
+    static css(node, property){
```
Comment (alexeypt) [inline Task7/src/index.js]: I am a bit confused how it works. You can pass `property` but you can't pass `value`.

See https://api.jquery.com/css/

File: Task7/src/index.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+console.log(jQuery.children(".demo-container", 'ul'));
+console.log(jQuery.attr(".demo-box", 'something'));
+
+console.log(jQuery.css('body', 'background-color'));
+
+
+jQuery.click('body', function() {
+    alert("addEventListener"); 
+  });
+
```
Comment (alexeypt) [inline Task7/src/index.js]: Please, check whether chaining works. I tried to call an example from task and there are lots of errors.

Please, be sure that such simplified example works:
```
new jQuery(p')
    .foreach(function (element) {
        console.log(element);
    })
    .append('<div>Test</div>')
    .addClass('my-super-class')
    .css('backgroundColor', 'rebeccapurple');
```

File: Task7/src/index.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            elements[i].parentNode.removeChild(elements[i]);
+        }        
+    };
+
+    text(){
+        return jQuery.text(this._string);
+    }
+
+    static text(node){
+        let element = document.querySelector(node);
```
[the remark this answers is not in the document]
Comment (Lapochka15) [reply Task7/src/index.js]: Done

File: Task7/src/index.js (hunk 1 above)
Comment (Lapochka15) [reply Task7/src/index.js]: Corrected

File: Task7/src/index.js (hunk 2 above)
Comment (Lapochka15) [reply Task7/src/index.js]: Corrected

File: Task7/src/index.js (hunk 3 above)
Comment (Lapochka15) [reply Task7/src/index.js]: Checked, chaining is working

File: Task7/src/index.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            callback.call(null, elements[i]);
+        }
+    }
+
+    toggle(){
+        jQuery.toggle(this._string);
+        return this;
+    }
+
+    static toggle(selector){
```
[the remark this answers is not in the document]
Comment (Lapochka15) [reply Task7/src/index.js]: Corrected

File: Task7/src/index.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        }
+        let result = [];
+        if (!selector)
+            return element.childNodes;
+
+        for(let i = 0; i < element.childNodes.length; i++ ){            
+            if (element.childNodes[i].nodeType == 1 && 
+                (element.childNodes[i].matches == "undefined"
+                ? element.childNodes[i].matches(selector)
+                :element.childNodes[i].msMatchesSelector(selector)
```
Comment (Happi-cat) [inline Task7/src/index.js]: better to move into separate function and replace with `matches(element.childNodes[i])`

**Pull request merged**